get_azure_blob.go

```diff
@@ -0,0 +1,307 @@
+	//if err != nil {
+	//	return 0, err
+	//}
+	//
+	//container := client.GetContainerReference(containerName)
+	//
+	//containerReference := storage.GetContainerReference(containerName)
+	//blobReference := containerReference.GetBlobReference(c.keyName)
+	//options := &storage.GetBlobOptions{}
+	//
+	//// List the object(s) at the given prefix
+	//params := storage.ListBlobsParameters{
+	//	Prefix: blobPath,
+	//}
+	//resp, err := container.ListBlobs(params)
+	//if err != nil {
+	//	return 0, err
+	//}
+	//
+	//for _, b := range resp.Blobs {
+	//	// Use file mode on exact match.
+	//	if b.Name == blobPath {
+	//		return ClientModeFile, nil
+	//	}
+	//
+	//	// Use dir mode if child keys are found.
+	//	if strings.HasPrefix(b.Name, blobPath+"/") {
+	//		return ClientModeDir, nil
+	//	}
+	//}
+	//
+	//// There was no match, so just return file mode. The download is going
+	//// to fail but we will let Azure return the proper error later.
+	//return ClientModeFile, nil
+	//ClientModeFile := nil
+
+	// From the Azure portal, get your storage account name and key and set environment variables.
+	//accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+	//if len(accountName) == 0 || len(accountKey) == 0 {
+	//	log.Fatal("Either the AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+	//}
+	//
+	//// Create a default request pipeline using your storage account name and account key.
+	//credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	//if err != nil {
+	//	log.Fatal("Invalid credentials with error: " + err.Error())
+	//}
+	//p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+	//
+	//// Create a random string for the quick start container
+	//containerName := fmt.Sprintf("quickstart-%s", randomString())
+	//
+	//// From the Azure portal, get your storage account blob service URL endpoint.
+	//URL, _ := url.Parse(
+	//	fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
+	//
+	//// Create a ContainerURL object that wraps the container URL and a request
+	//// pipeline to make requests.
+	//containerURL := azblob.NewContainerURL(*URL, p)
+	//
+	//// Create the container
+	//fmt.Printf("Creating a container named %s\n", containerName)
+	//ctx := context.Background() // This example uses a never-expiring context
+	//_, err = containerURL.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
+	//handleErrors(err)
+	//
+	//
+	//
+	//
+	//// Here's how to download the blob
+	//downloadResponse, err := blobURL.Download(ctx, 0, azblob.CountToEnd, azblob.BlobAccessConditions{}, false)
+	//
+	//// NOTE: automatically retries are performed if the connection fails
+	//bodyStream := downloadResponse.Body(azblob.RetryReaderOptions{MaxRetryRequests: 20})
+	//
+	//// read the body into a buffer
+	//downloadedData := bytes.Buffer{}
+	//_, err = downloadedData.ReadFrom(bodyStream)
+	//handleErrors(err)
```

Comment:
Should we keep this commented code ? Is this a Work In Progress ? ( Ditto throughout )

Reply:
Is it because the blog getter doesn't have the capability to download folders ?